app/main_proxy.go

```diff
@@ -58,7 +58,9 @@ func (s *fakeListener) Close() error {
 	case <-s.closed:
 		return nil
 	default:
-		close(s.closed)
+		if s.closed != nil {
```

Comment:
I haven't set `proxy_server` field in my `config.json` file, and the `s.closed` is initialized in `Server()` function.
Another good amendment is to initialize the `s.closed` in init function, for instance.